### ISSUE TITLE
Lehet-e a keyboard repeat delay-t csökkenteni?

### ISSUE THREAD

**Issue by self-driven** [SzFMV2017-Osz/AutomatedCar#198]
Kormányzásnál elég zavaró, hogy lenyomott billentyű automatikus ismétlése előtt "nagy" a késleltetés. Ha lehetne ezt az értéket kódból csökkenteni realisztukusabb lenne az irányítás.

**Comment by danijanos** [comment]
Szia @self-driven, persze! megcsináljuk.
ebben a sorban van egy 100ms -es érték:

https://github.com/SzFMV2017-Osz/AutomatedCar/blob/96f63ee9adda0ac19ddd611c12075b5a8c05f019/src/main/java/hu/oe/nik/szfmv/inputinterface/UserInputHandler.java#L139

már nem emlékszem miért írtam oda... de javítom, kösz, hogy szóltál! 👍

**Comment by self-driven** [comment]
Lehet, hogy ez az érték is beleszól, de próbaképpen átírtam és nem javult
meg.
A billentyűzet ismétlés előtti késleltetés tudtommal valami hardver közeli
érték, mely szerintem konkrétan a billentyűzetben van tárolva. Ezt régebben
a gép bios-ában is be lehetett állítani, az oprendszer is támogatja a
módosítását, és valahogy biztos elérhető, hogy kódból is lehessen
változtatni.
Ha lenyomod valamelyik "kormány" billentyűt és nézed, hogy a dashboard-on
hogyan változik a kijelzett érték, az látszik, hogy "elég sokat" vár
mielőtt elkezdené az érték folyamatos növelését.

2017. november 12. 13:18 Dani János írta, <notifications@github.com>:

**Comment by danijanos** [comment]
@self-driven , igen, közben rájöttem, hogy az az érték azt álltja, hogy mennyi idő múlva kezdjen el visszaállni minden pedál state a kezdeti pozíciójába. Szóval igazából semmi köze nincs ahhoz, amit te szeretnél. bocs.

egyébként a hívás lánc kb így néz ki:
1. gomblenyomás után, a lenyomott gomb "kódja" bekerül egy listába (ez a szimultán lenyomások miatt kellett)
2. ha a kormányzás gomb van megnyomva, akkor az osztály a saját belő változóját felülírja egy másik objektumból hívott függvény segítségével (ebben csak két értékvizsgálat van, és egy értékadás)
3. majd ez az új érték úgy kerül a buszra, hogy az ISystemComponent által előírt loop() metódusban az objektum rádobja a saját (megváltozott) belső változóját.

szóval ez is minimum 3-4 függvényhívás, mire a konkrét érték a gombnyomás utána valahová kerül. Plusz az, hogy utána mi történik, mire kiíródik a dashboardra, azt nem tudom.

nem tudom, hogy most mit csinálsz vagy mi a cél, de nem lehet, hogy ez a beállítás segít ezen?:

https://computing.which.co.uk/hc/en-gb/articles/115002584785-Speed-up-or-slow-down-a-keyboard

https://superuser.com/questions/388160/keyboard-repeat-rate-repeat-delay-values-in-win7

**Comment by self-driven** [comment]
Ezek az időtartamok amiket felsoroltál összességében elenyészőek a
billentyűzet repeat delay beállításához képest.

Megnéztem az első linket amit küldtél és az erre vonatkozik, de szerintem
nem a felhasználó hatáskörébe tartozik, hogy az alkalmazás megfelelően
kezelje a billentyűzet ismétlési időzítéseit.

https://stackoverflow.com/questions/13043229/keypressed-event-slow-on-first-repeat

2017. november 12. 16:19 Dani János írta, <notifications@github.com>:

**Comment by danijanos** [comment]
@self-driven oké, de szerintem még mielőtt egy plusz Timer szálat futtatunk, gondoljuk végig, hogy ha csak az a cél, hogy gyorsabban forduljon az autó, akkor ezt egyszerűbben is megoldhatjuk. Hiszen, ha jól értem az érték gyorsabb változását szeretnéd elérni. Ezt nem csak az ismétlődés idejének csökkentésével tehetjük meg, ergo azzal, hogy adott időintervallumon belül többször hívódik meg az értéknövelő függvény, hanem az elfordítás finomságának, skálájának, felbontásának növelésével is. Magyarul, ha például ezt a mínusz 1 értéket átírod:

https://github.com/SzFMV2017-Osz/AutomatedCar/blob/96f63ee9adda0ac19ddd611c12075b5a8c05f019/src/main/java/hu/oe/nik/szfmv/inputinterface/CarComponentStateCalculator.java#L8

mondjuk 5-re.

Ha így működik a dolog, írj és megírom hogy ez egy külön változóban szerepeljen és a jobbra illetve balra kormányzásnál is ugyanannyival módosuljon a kormánykerék állása.

pl:
```
private int steeringWheelRotationResolution = 5;
```

**Comment by self-driven** [comment]
Nem az érték gyorsabb változását szeretném elérni, azt be tudtam állítani
miután átolvastam a kódotokat.
Azt szeretném elérni, hogy a billentyűzet lenyomáskor visszaadott első kód
után ne teljen el sok idő, míg ugyanazt a kódot elkezdi megfelelő ütemben
ismételgetni.

2017. november 14. 10:52 Dani János írta, <notifications@github.com>:
